examples/misc_custom_controls.html

```diff
@@ -0,0 +1,112 @@
+        <li>Wheel-click + drag : rotates the camera around its target</li>
+        <li>Ctrl + left-click + drag : rotates the camera around its position</li>
+        <li>Mouse wheel up/down : zooms in/out</li>
+        <li>Double left-click : zoom in</li>
+        <li>Double right-click : zoom out</li>
+        <li>Right-click + drag up/down : zoom in/out</li>
+    </ul>
+</div>
+<div id="viewerDiv"></div>
+<span id="divScaleWidget"> Scale </span>
+
+<script src="../dist/itowns.js"></script>
+<script src="../dist/debug.js"></script>
+<script src="js/GUI/LoadingScreen.js"></script>
+<script src="js/GUI/GuiTools.js"></script>
+<script src="js/Scale.js"></script>
+<script type="text/javascript">
+    /* global itowns, setupLoadingScreen, GuiTools, debug */
+
+    const viewerDiv = document.getElementById('viewerDiv');
+    const placement = {
+        coord: new itowns.Coordinates('EPSG:4326', 2.351323, 48.856712),
+        range: 25000000,
+    }
+    const view = new itowns.GlobeView(viewerDiv, placement);
+    setupLoadingScreen(viewerDiv, view);
+
+    // Add imagery and elevation layers
+    itowns.Fetcher.json('./layers/JSONLayers/Ortho.json').then(function (config) {
+        config.source = new itowns.WMTSSource(config.source);
+        view.addLayer(
+            new itowns.ColorLayer('Ortho', config),
+        );
+    })
+    function addElevationLayerFromConfig(config) {
+        config.source = new itowns.WMTSSource(config.source);
+        view.addLayer(
+            new itowns.ElevationLayer(config.id, config),
+        );
+    }
+    itowns.Fetcher.json('./layers/JSONLayers/WORLD_DTM.json').then(addElevationLayerFromConfig);
+    itowns.Fetcher.json('./layers/JSONLayers/IGN_MNT_HIGHRES.json').then(addElevationLayerFromConfig);
+
+    const customControls = {
+        // Disable pan movement
+        PAN: {
+            enable: false,
+        },
+        // Change the key bindings for globe rotation
+        MOVE_GLOBE: {
+            mouseButton: itowns.THREE.MOUSE.LEFT,
+        },
+        // Change the key bindings for orbit movement (rotation around the camera target)
+        ORBIT: {
+            mouseButton: itowns.THREE.MOUSE.MIDDLE,
+            keyboard: undefined,
+        },
+        // Change the key bindings for dolly movement
+        DOLLY: {
+            mouseButton: itowns.THREE.MOUSE.RIGHT,
+        },
+        // Change the key bindings for panoramic movement (rotation around the camera position)
+        PANORAMIC: {
+            mouseButton: itowns.THREE.MOUSE.LEFT,
+            keyboard: 17,  // keyCode for the ctrl key
+        },
+        // Allow travel out movement when double right-clicking
+        TRAVEL_OUT: {
+            enable: true,
+            trigger: 'dblclick-right',
+        },
+    }
+
+    // Modify view's control to be set as the custom controls we just defined
+    view.addEventListener(itowns.GLOBE_VIEW_EVENTS.GLOBE_INITIALIZED, function () {
```

Comment:
could you use a new instance of `StateControls` or directly modify the current instance...

Reply:
I added a method in `StateControls` : `setFromOptions` which allows passing all the custom options to `controls.states`.